### PR TITLE
Rebased ezdl command with more sites supported

### PR DESCRIPTION
**Pull request by Blatzar**
Should now not error on watch

### CODE REVIEW

**Comment by Blatzar** [comment]
TODO: dub/sub flag should probably be easy to implement for ezdl, but I'll solve that in a separate PR

File: anime_downloader/commands/ezdl.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+            logger.debug('Current provider: {}'.format(provider))
+            # TODO: Replace by factory
+            cls = get_anime_class(anime_url)
+
+            # To make the downloads use the correct name if URL:s are used.
+            real_provider = cls.sitename if cls else provider
+            # This will allow for animeinfo metadata in filename and one filename for multiple providers.
+            rep_dict = {
+                'animeinfo_anime_title': util.slugify(info.title),
```
Comment (MichaelDavidJohnson) [inline anime_downloader/commands/ezdl.py:116]: for readability I would prefer that `utils.slugify(info.title)` and variants were initialised outside of the dictionary with an informative variable name. The comment doesn't cut it for ease of initial reading

File: anime_downloader/commands/watch.py [hunk 2]
```diff
@@ -67,7 +67,7 @@ def command(anime_name, new, update_all, _list, quality, remove,
         else:
             query = click.prompt('Enter a anime name or url', type=str)
 
-        url = util.search(query, provider)
+        url, _ = util.search(query, provider)
```
Comment (MichaelDavidJohnson) [inline anime_downloader/commands/watch.py:70]: does this cause any unintended behaviour that you have noticed?

File: anime_downloader/config.py [hunk 3]
```diff
@@ -25,120 +25,137 @@
...
             ]
         },
+        'kissanime': {
+            'version':'subbed'
+        },
+        'animeonline360': {
+            'version':'subbed'
+        }
     }
 }
```
Comment (MichaelDavidJohnson) [inline anime_downloader/config.py:161]: I actually can't believe you went and changed those.. absolute madlad

File: anime_downloader/commands/ezdl.py [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    help='Ratio used for the auto select in search. 100 means it only auto selects on complete matches. 0 auto selects regardless of how similar the result is.',
+    default=50
+)
+@click.option(
+    '--url', '-u', type=bool, is_flag=True,
+    help="If flag is set, prints the stream url instead of downloading")
+@click.option("--skip-fillers", is_flag=True, help="Skip downloading of fillers.")
+
+@click.pass_context
+def command(ctx, anime_url, episode_range, player, quality,
```
[the remark this answers is not in the document]
Comment (Blatzar) [reply anime_downloader/commands/ezdl.py:63]: Copied from dl, pretty sure it's kinda standard since discord also uses that.

[the remark this answers is not in the document]
Comment (MichaelDavidJohnson) [reply anime_downloader/commands/ezdl.py:63]: Yeah, it requires a google. Fair enough

File: anime_downloader/commands/watch.py (hunk 2 above)
Comment (Blatzar) [reply anime_downloader/commands/watch.py:70]: No, I literally see no reasons why this would cause any problems

File: anime_downloader/commands/ezdl.py [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+    # Borrows some config from the original dl command.
+    fallback_qualities = Config['dl']['fallback_qualities']
+    download_dir = Config['dl']['download_dir']
+    quality = Config['dl']['quality'] if not quality else quality
+    url = Config['dl']['url'] if not url else url
+    external_downloader = Config['dl']['external_downloader']
+    skip_download = Config['dl']['skip_download']
+    chunk_size = Config['dl']['chunk_size']
+    speed_limit = Config['dl']['speed_limit']
+
```
[the remark this answers is not in the document]
Comment (Blatzar) [reply anime_downloader/commands/ezdl.py:76]: I see no issues, a whitespace to separate ezdl config reading to dl config reading. You're overanalyzing.

File: anime_downloader/commands/watch.py (hunk 2 above)
Comment (MichaelDavidJohnson) [reply anime_downloader/commands/watch.py:70]: Yeah, it doesn't. This was commented before testing.. just wanted to be made aware

**Comment by MichaelDavidJohnson** [comment]
BUG: Calling a specific broken provider causes an error and does not handle the iteration to the next provider in the list as intended.

File: anime_downloader/sites/animeonline.py [hunk 6]
```diff
@@ -16,6 +16,10 @@ def search(cls, query):
                 SearchResult(
                     title = i['title'],
                     url = i['url'],
+                    meta_info = {
```
Comment (vn-ki) [inline anime_downloader/sites/animeonline.py:19]: What is this meta info exactly?

File: anime_downloader/util.py [hunk 7]
```diff
@@ -78,35 +78,69 @@ def format_search_results(search_results):
...
+                         ' Please input a number less than {}'.format(
+                             len(search_results), len(search_results)+1))
+            val = False
+            continue
+        break
 
     logger.info('Selected {}'.format(title))
 
-    return url
+    return url, val
```
Comment (vn-ki) [inline anime_downloader/util.py:122]: Why not return search result?

File: anime_downloader/commands/ezdl.py [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+@click.option(
+    '--episodes', '-e', 'episode_range', metavar='<int>:<int>',
+    help="Range of anime you want to download in the form <start>:<end>")
+@click.option(
+    '--play', 'player', metavar='PLAYER',
+    help="Streams in the specified player")
+@click.option(
+    '--quality', '-q', type=click.Choice(['360p', '480p', '720p', '1080p']),
+    help='Specify the quality of episode. Default-720p')
+@click.option(
```
Comment (vn-ki) [inline anime_downloader/commands/ezdl.py:30]: This can be removed, we don't use it now a days.

File: anime_downloader/commands/ezdl.py [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+    '--force-download', '-f', is_flag=True,
+    help='Force downloads even if file exists')
+@click.option(
```
Comment (vn-ki) [inline anime_downloader/commands/ezdl.py:33]: Is file format required?

File: anime_downloader/commands/ezdl.py [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+    '--provider',
+    help='The anime provider (website) for search.',
+    type=click.Choice(sitenames)
+)
+@click.option(
+    '--disable-ssl',
+    is_flag=True,
+    help='Disable verifying the SSL certificate, if flag is set'
+)
+@click.option(
```
Comment (vn-ki) [inline anime_downloader/commands/ezdl.py:48]: Isnt this automated? Why is this required?

Comment (MichaelDavidJohnson) [reply anime_downloader/commands/ezdl.py:48]: He still wants to give the user these options citing debugging as a reason.

Comment (MichaelDavidJohnson) [reply anime_downloader/commands/ezdl.py:48]: It seems he wants to give the user basically the same options as a dl command. Citing debugging as the main reason. I assume @vn-ki the command you are questioning is `-c`?

File: anime_downloader/sites/animeonline.py (hunk 6 above)
Comment (Blatzar) [reply anime_downloader/sites/animeonline.py:19]: Metadata which doesn't show up in search.

File: anime_downloader/util.py (hunk 7 above)
Comment (Blatzar) [reply anime_downloader/util.py:122]: To save the result for future episodes. You wouldn't wanna enter your choice 12 times if there's 12 episodes.

File: anime_downloader/commands/ezdl.py (hunk 8 above)
Comment (Blatzar) [reply anime_downloader/commands/ezdl.py:30]: Sure

File: anime_downloader/commands/ezdl.py (hunk 9 above)
Comment (Blatzar) [reply anime_downloader/commands/ezdl.py:33]: No, I always use config for that since it's too long. Removing it.

File: anime_downloader/commands/ezdl.py (hunk 10 above)
Comment (Blatzar) [reply anime_downloader/commands/ezdl.py:48]: Ssl certificate? Legacy code I was spooked to remove, can definitely remove.
-c? Remnants from before I had ratio added as flag, -r 101 is more consistent from a debugging perspective so I'm OK with removing it.
